src/psyclone/tests/gocean1p0_test.py

```diff
@@ -973,6 +975,21 @@ def test00p1_kernel_wrong_meta_arg_count():
               api="gocean1.0")
 
 
+def test00p1_invoke_kernel_using_const_scalar():  # pylint:disable=invalid-name
+    '''Check that using a const scalar as parameter works.'''
+    filename = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                            "test_files", "gocean1p0",
+                            "test00.1_invoke_kernel_using_const_scalar.f90")
+    _, invoke_info = parse(filename, api="gocean1.0")
+    out = str(PSyFactory(API).create(invoke_info).gen)
+    # Old versions of PSyclone tried to declare '0' as a variable:
+    # REAL(KIND=wp), intent(inout) :: 0
+    # Make sure this is not happening anymor
+    import re
+    assert re.search(r"\s*real.*:: *0", out, re.I) is None
+    assert re.search(r"\s*real.*:: *real_val", out, re.I) is not None
```

Comment:
:-) When I said 'real' I meant try testing cases where both a real quantity e.g. "0.0_dp" and an integer quantity, e.g. "1" are passed by value. I realise that, on looking at this again the thing being passed as "0" is in fact a real variable so my request was confusing. Does it make sense now?

Reply:
*grin* the 'by value' caused me confusion. I added an integer parameter to the new test function, and passed both a constant 0 as real and int in. I left the test with by-value in place, can't really hurt as far as I can tell.